### DISCUSSION TITLE
Task 3 Assignment 2

### DISCUSSION THREAD

**Discussion by CraigButikofer** [USU-ITLS-x265-23-summer/class-repo#5] [Week 1 - Discussions]
I'm not sure how to show that I have completed this assignment. It looks like it could be considered complete just by reading through it, but there is a list of optional activities that seem like they are important for understanding how to proceed with the course. I am not sure how or where to designate that I have completed the initial task of reading the set up content. It appears that a request cannot be pulled until an action is created that changes my item from the initial repository, I tried creating a "fork" so that I could "pull a request" but I have no idea how to create it in the context that I am being given. I can't seem to get the changes to stick or carry so I can't "pull a request". I'm wondering, if we could get some videos that walk us through the beginning steps of what is expected? There are just so many places to click and so many tabs that are opening and I'm having trouble understanding how everything connects.

**Comment by metasean** [comment]
Bear with me for a few moments.

Several of these tasks, including this one, are just an introduction to some of the important moving parts of git and github, and they're done in an isolated area with just you involved.  Normally when you're using PRs there are other people involved.  (Think about this exercise like a factory tour showing you each task that can be done.  But on a realistic day, you'd only be doing one of the tasks.  This tour helps you know what the different tasks are, how to do them, and when you need to do them.  Then you'll have some time to practice them on the course repo.)

(a) Looking at your repo, you're actually doing a great job of following along, and it also looks like you're doing some exploring. 👍 

(b) In step 3, you successfully created a Pull Request. 👍 
<img width="400" alt="Screen Shot 2023-05-09 at 4 06 05 PM" src="https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/7552754/3756d387-b42f-4a06-8417-aff83508d034">

(c) Since there's only one pull request, when you click the tab highlighted in the screenshot above, you should see one entry in the Pull Requests list (named "Feedback"), when you click on that, you should see a screen with a bunch of information about that particular PR.  Immediately below the name and id of the PR (e.g. "Feedback #1") you'll see the line:  

"[github-classroom](https://github.com/apps/github-classroom) wants to merge 2 commits into [feedback](https://github.com/USU-ITLS-x265-23-summer/introduction-to-github-CraigButikofer/tree/feedback) from [main](https://github.com/USU-ITLS-x265-23-summer/introduction-to-github-CraigButikofer)

This means that a user by the name of `github-classroom` (which is just a bot in this tutorial) is trying to merge the `main` branch, into the `feedback` branch (which is opposite of what you'd normally want to do, you normally are merging into `main`)


Below that you have a description of the change, and commits on that branch.  You can also see the list of commits via the "Commits" tab.  There's a "Checks" tab that we won't be using in this class.  And finally, there's a "Files Changed" tab.

The two important parts for what you'll be doing is the description and files changed tabs.  Take a look at the "Files Changed" in your PR, and you should be able to tell what's different between the two branches.  If you approve of the changes, then you go back to the conversation tab, scroll down and merge the pull request.  

In the [next assignment](https://github.com/USU-ITLS-x265-23-summer/class-repo/blob/main/Tasks/Task-3.md#assignment-4) you're going to move from your practice repos, to our class repo.  You're going to do each of the different things that you've learned so far, but in a repo with other people.  For example, derekscholes has already cloned the class-repo, created a new branch in his copy, and has made a PR to merge his changes into the official class-repo repo.  

Outside of these introductory lessons, you shouldn't be merging your own PRs.  You make a PR and someone else reviews it and agrees (or not) to merge it into the `main` branch on the `origin` repo.  Likewise, when someone else has made a PR, you can review it and if you agree with the changes, then you can merge them.

Because I made some live changes (something you should try to avoid, but occasionally are necessary), in this case,  derekscholes' `patch-1` branch on his copy of the class-repo doesn't match the current class-repo, so there's an extra step, but it should be pretty obvious for whoever tries to merge his PR.

Right now, it feels really, really confusing, convoluted, and very jargony.  

The jargon actually does start to make a bit more sense as you work with it and have conversations.  

There are several ways to use git commands.  The primary reason that I'm trying to make sure y'all can use the GitHub Classrooms is because it gives you free (within reason) access to a tool that will make a whole lot of things easier -including git and github— but if you don't know how the different parts of the factory work, you can get yourself in a world of hurt.  (And sadly, it wouldn't be as tasty as this [factory fiasco](https://www.youtube.com/watch?v=AnHiAWlrYQc)

There's one more exercise (Assignment 5) that should help you understand the PR process a little bit better.  And you'll be using what you learn in the class-repo in Assignment 6.

If it doesn't start to make some sense by then, let me know!

**Comment by bayleeriley** [comment]
Thank you for asking this Craig. I was wondering the same thing!
